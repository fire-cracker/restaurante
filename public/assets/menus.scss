.menu-jumbotron {
  padding-top: 90px;
  padding-bottom: 70px;
  margin-bottom: 0;
  font-family: 'Montserrat', Tahoma, Geneva, sans-serif;
  .menu-container {
    .container-body {
      .menu-header{
        text-align: center;
        h4{
          font-weight: 600;
          font-size: 1rem;
        }
        h2{
          font-family: var(--font-family-bellefair);
          font-weight: 400;
          font-size: 3rem;
        }
        .divider{
          & .line {
            border-bottom: 1px solid var(--darkkhaki);
          }
          .icon{
            margin: 0;
          }
        }
      }
      .navs-container {
        font-size: 1em;
        .nav {
          .nav-item {
            text-align: center;
            // width: 75%;
            .nav-link {
              border: none;
              box-shadow: 0 0 6px 0 rgba(26, 42, 58, 0.03);
              font-size: 16px;
              color: #666;
              font-weight: 500;
              &.active {
                background: var(--darkkhaki) !important;
              }
              p {
                color: #1b2c3d;
                font-size: 17px;
                line-height: 24px;
                margin-left: 0;
              }
            }
            .nav-link:hover {
              background: var(--darkkhaki) !important;
            }
          }
        }
      }
      .menu-body {
        .row {
          .media {
            img {
              border: 5px solid var(--white);
              box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.07);
            }
            .media-body {
              line-height: 26px;
              h5 {
                border-bottom: 1px dashed #cdcdcd;
                font-size: 24px;
              }
              p {
                color: #4d5257;
                font-size: 14px;
              }
            }
          }
        }
      }
    }
  }
}

@media screen and (max-width: 1000px) {
  .menu-container {
    .container-body {
      .navs-container {
        margin: auto !important;
        .nav {
          .nav-col {
            margin-bottom: 2rem;
            .nav-item {
              width: auto !important;
              margin: auto !important;
            }
          }
        }
      }
      .menu-body {
        .row {
          width: 100%;
          margin: auto;
          .media {
            align-items: center;
            img {
              margin: 0 !important;
            }
            .media-body {
              width: auto;
              padding-left: 1rem;
            }
          }
        }
      }
    }
  }
}

@media screen and (max-width: 600px) {
  .menu-container {
    .container-body {
      .navs-container {
        margin: auto !important;
        .nav {
          .nav-col {
            margin-bottom: 1rem;
            .nav-item {
              width: auto !important;
              margin: auto !important;
            }
          }
        }
      }
      .menu-body {
        .row {
          width: 100%;
          margin: auto;
          .media-col {
            padding: 0;
            .media {
              flex-direction: column;
              align-items: center;
              img {
                margin: 0 !important;
                width: 100%;
                height: auto;
              }
              .media-body {
                width: 100%;
                padding-left: 0;
              }
            }
          }
        }
      }
    }
  }
}

@media only screen and (min-width: 1000px) {
  .menu-container {
    padding: 0 !important;
  }
}
