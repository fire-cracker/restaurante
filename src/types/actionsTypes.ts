export interface Action<T> {
  type: ActionType;
  payload?: T;
}

export enum ActionType {
  SIGNUP_REQUEST_PENDING = 'SIGNUP_REQUEST_PENDING',
  SIGNUP_REQUEST_SUCCESS = 'SIGNUP_REQUEST_SUCCESS',
  SIGNUP_REQUEST_FAILED = 'SIGNUP_REQUEST_FAILED',
  
  LOGIN_REQUEST_PENDING = 'LOGIN_REQUEST_PENDING',
  LOGIN_REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS',
  LOGIN_REQUEST_FAILED = 'LOGIN_REQUEST_FAILED',

  GET_PROFILE_REQUEST_PENDING = 'GET_PROFILE_REQUEST_PENDING',
  GET_PROFILE_REQUEST_SUCCESS = 'GET_PROFILE_REQUEST_SUCCESS',
  GET_PROFILE_REQUEST_FAILED = 'GET_PROFILE_REQUEST_FAILED',

  LOGOUT_REQUEST_SUCCESS = 'LOGOUT_REQUEST_SUCCESS',

  SET_USER_STATE = 'SET_USER_STATE'
}