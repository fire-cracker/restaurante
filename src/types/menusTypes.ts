export interface IMenu {
  id: number
  name: string
  image: string
  price: number
  recipe: string
  type: string
}

export interface IMenuState {
  menus: IMenu | null
  count: number
  fetching: boolean
  fetched: boolean
}
