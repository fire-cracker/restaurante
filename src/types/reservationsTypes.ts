export interface IReservation {
  id: number
  userId: string
  date: string
  time: string
  type: string
  price: number
  persons: number
}
export interface INewReservation {
  date: string
  time: string
  type: string
  persons: number
}

export interface IReservationState {
  reservation: IReservation | null
  fetching: boolean
  fetched: boolean
}

export interface IStripeCharge {
  id: string
  object: string
  amount: 2000
  paid: boolean
  receipt_email: string
  receipt_number: string
  receipt_url: string
  status: string
}
