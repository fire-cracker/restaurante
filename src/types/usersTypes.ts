export interface IUser {
  id: string
  username: string
  email: string
  role: 'customer' | 'admin'
}

export interface IUserState {
  user: IUser | null
  logingIn: boolean
  isLoggedIn: boolean
}