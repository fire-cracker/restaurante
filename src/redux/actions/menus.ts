import { Dispatch } from 'redux'
import { toast } from 'react-toastify'

import axios from '../../utils/axiosConfig'
import { ActionType, Action } from '../../types/actionsTypes'
import { IMenu } from '../../types/menusTypes'

const {
  GET_MENUS_REQUEST_PENDING,
  GET_MENUS_REQUEST_SUCCESS,
  GET_MENUS_REQUEST_FAILED
} = ActionType

const api = process.env.REACT_APP_API_URL

export const getMenusRequestPending = (): Action<any> => ({
  type: GET_MENUS_REQUEST_PENDING
})

export const getMenusRequestSuccess = (data: {
  menus: IMenu[]
  count: number
}): Action<{ menus: IMenu[]; count: number }> => ({
  type: GET_MENUS_REQUEST_SUCCESS,
  payload: data
})

export const getMenusRequestFailed = (): Action<any> => ({
  type: GET_MENUS_REQUEST_FAILED
})

export const getMenus = () => async (
  dispatch: Dispatch<Action<any>>
): Promise<{ menus: IMenu[]; count: number }> => {
  try {
    dispatch(getMenusRequestPending())
    const {
      data: { data }
    } = await axios.get(`${api}/menus`)
    dispatch(getMenusRequestSuccess(data))
    return data
  } catch (error) {
    dispatch(getMenusRequestFailed())
    toast.error(error.message)
    throw error
  }
}
