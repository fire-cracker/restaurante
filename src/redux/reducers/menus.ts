import { ActionType, Action } from '../../types/actionsTypes'
import { IMenuState } from '../../types/menusTypes'

const {
  GET_MENUS_REQUEST_PENDING,
  GET_MENUS_REQUEST_SUCCESS,
  GET_MENUS_REQUEST_FAILED
} = ActionType

const initialState: IMenuState = {
  menus: null,
  count: 0,
  fetching: false,
  fetched: false
}
export default (state = initialState, action: Action<any>): IMenuState => {
  switch (action.type) {
    case GET_MENUS_REQUEST_PENDING:
      return {
        ...state,
        fetching: true
      }

    case GET_MENUS_REQUEST_SUCCESS:
      return {
        ...state,
        ...action.payload,
        fetching: false,
        fetched: true
      }

    case GET_MENUS_REQUEST_FAILED:
      return {
        ...state,
        fetching: false,
        fetched: false
      }

    default:
      return state
  }
}
