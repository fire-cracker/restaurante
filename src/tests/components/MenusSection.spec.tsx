import React, { RefObject } from 'react'
import { render, cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import MenusSection from '../../components/Menus'
import { IMenu } from '../../types/menusTypes'
import { menus } from '../mocks/menus.mock'

jest.mock('../../utils/axiosConfig')
jest.mock('../../redux/actions/menus')

const createMockStore = configureMockStore([thunk])

const store = createMockStore({})

interface IProps {
  menuRef?: RefObject<HTMLInputElement>
  getMenus: () => Promise<{ menus: IMenu[]; count: number }>
  menus?: IMenu[]
}
describe('Header', () => {
  const defaultProps: IProps = {
    menuRef: React.createRef(),
    getMenus: jest.fn().mockResolvedValue({menus}),
    menus: menus
  }

  const setup = (newProps?: any) => {
    const props = { ...defaultProps, ...newProps }
    return render(
      <Provider store={store}>
        <MenusSection {...props} menus={menus}/>
      </Provider>
    )
  }

  afterAll(() => {
    jest.clearAllMocks()
    cleanup()
  })

  test('renders MenusSection component', () => {
    const wrapper = setup()
    expect(wrapper).toMatchSnapshot()
    const { getByText } = wrapper
    const breakfastLink = getByText('Breakfast')
    expect(breakfastLink).toBeInTheDocument()
    const lunchLink = getByText('Lunch')
    expect(lunchLink).toBeInTheDocument()
    const dinnerLink = getByText('Dinner')
    expect(dinnerLink).toBeInTheDocument()
  })

  test('should render menus if getmenu request it successful', () => {
    const wrapper = setup()
    // wrapper.debug()
  })
})
