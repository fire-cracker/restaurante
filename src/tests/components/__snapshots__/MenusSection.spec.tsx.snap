// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders MenusSection component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
      >
        <div
          class="menu-container m-auto container"
        >
          <div
            class="container-wrapper px-0 row"
          >
            <div
              class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
            >
              <div
                class="menu-header mb-4 flex-column justify-content-center align-items-center row"
              >
                <div
                  class="col"
                >
                  <h4
                    class="text-darkkhaki"
                  >
                    FOOD AT FIRST SIGHT
                  </h4>
                </div>
                <div
                  class="col"
                >
                  <h2>
                    Our Restaurant Menu
                  </h2>
                </div>
                <div
                  class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                >
                  <p
                    class="line my-2 mx-1"
                  >
                     
                  </p>
                  <p
                    class="icon"
                  >
                    <img
                      alt="entwined"
                      class="fa fa-cutlery"
                      src="/images/entwined.svg"
                    />
                  </p>
                  <p
                    class="line my-2 mx-1"
                  >
                     
                  </p>
                </div>
              </div>
              <div
                class="navs-container mb-4 w-100 row"
              >
                <div
                  class="w-100 justify-content-between nav"
                  role="tablist"
                >
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="true"
                        class="py-4 px-3 bg-white nav-link active"
                        data-rb-event-key="breakfast"
                        href="#"
                        role="tab"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/breakfast.svg"
                        />
                        <p>
                          Breakfast
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="lunch"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/lunch.svg"
                        />
                        <p>
                          Lunch
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-4 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="dinner"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/dinner.svg"
                        />
                        <p>
                          Dinner
                        </p>
                      </a>
                    </div>
                  </div>
                  <div
                    class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                  >
                    <div
                      class="mr-0 mt-1 mb-0 p-0 nav-item"
                    >
                      <a
                        aria-selected="false"
                        class="py-4 px-3 bg-white nav-link"
                        data-rb-event-key="drink"
                        href="#"
                        role="tab"
                        tabindex="-1"
                      >
                        <img
                          alt="breafast icon"
                          src="/images/drink.svg"
                        />
                        <p>
                          Drinks
                        </p>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="menu-body mx-0 pt-5 row"
              >
                <div
                  class="tab-content"
                >
                  <div
                    aria-hidden="false"
                    class="fade tab-pane active show"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="fade tab-pane"
                    role="tabpanel"
                  >
                    <div
                      class="row"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menu-jumbotron bg-ghostwhite px-0 jumbotron jumbotron-fluid"
    >
      <div
        class="menu-container m-auto container"
      >
        <div
          class="container-wrapper px-0 row"
        >
          <div
            class="container-body m-0 d-flex flex-column justify-content-center align-items-center container"
          >
            <div
              class="menu-header mb-4 flex-column justify-content-center align-items-center row"
            >
              <div
                class="col"
              >
                <h4
                  class="text-darkkhaki"
                >
                  FOOD AT FIRST SIGHT
                </h4>
              </div>
              <div
                class="col"
              >
                <h2>
                  Our Restaurant Menu
                </h2>
              </div>
              <div
                class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
              >
                <p
                  class="line my-2 mx-1"
                >
                   
                </p>
                <p
                  class="icon"
                >
                  <img
                    alt="entwined"
                    class="fa fa-cutlery"
                    src="/images/entwined.svg"
                  />
                </p>
                <p
                  class="line my-2 mx-1"
                >
                   
                </p>
              </div>
            </div>
            <div
              class="navs-container mb-4 w-100 row"
            >
              <div
                class="w-100 justify-content-between nav"
                role="tablist"
              >
                <div
                  class="nav-col p-0 col-lg-3 col-md-5"
                >
                  <div
                    class="mr-4 mt-1 mb-0 p-0 nav-item"
                  >
                    <a
                      aria-selected="true"
                      class="py-4 px-3 bg-white nav-link active"
                      data-rb-event-key="breakfast"
                      href="#"
                      role="tab"
                    >
                      <img
                        alt="breafast icon"
                        src="/images/breakfast.svg"
                      />
                      <p>
                        Breakfast
                      </p>
                    </a>
                  </div>
                </div>
                <div
                  class="nav-col p-0 col-lg-3 col-md-5"
                >
                  <div
                    class="mr-4 mt-1 mb-0 p-0 nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="py-4 px-3 bg-white nav-link"
                      data-rb-event-key="lunch"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <img
                        alt="breafast icon"
                        src="/images/lunch.svg"
                      />
                      <p>
                        Lunch
                      </p>
                    </a>
                  </div>
                </div>
                <div
                  class="nav-col p-0 col-lg-3 col-md-5"
                >
                  <div
                    class="mr-4 mt-1 mb-0 p-0 nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="py-4 px-3 bg-white nav-link"
                      data-rb-event-key="dinner"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <img
                        alt="breafast icon"
                        src="/images/dinner.svg"
                      />
                      <p>
                        Dinner
                      </p>
                    </a>
                  </div>
                </div>
                <div
                  class="nav-col mr-0 p-0 col-lg-3 col-md-5"
                >
                  <div
                    class="mr-0 mt-1 mb-0 p-0 nav-item"
                  >
                    <a
                      aria-selected="false"
                      class="py-4 px-3 bg-white nav-link"
                      data-rb-event-key="drink"
                      href="#"
                      role="tab"
                      tabindex="-1"
                    >
                      <img
                        alt="breafast icon"
                        src="/images/drink.svg"
                      />
                      <p>
                        Drinks
                      </p>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="menu-body mx-0 pt-5 row"
            >
              <div
                class="tab-content"
              >
                <div
                  aria-hidden="false"
                  class="fade tab-pane active show"
                  role="tabpanel"
                >
                  <div
                    class="row"
                  />
                </div>
                <div
                  aria-hidden="true"
                  class="fade tab-pane"
                  role="tabpanel"
                >
                  <div
                    class="row"
                  />
                </div>
                <div
                  aria-hidden="true"
                  class="fade tab-pane"
                  role="tabpanel"
                >
                  <div
                    class="row"
                  />
                </div>
                <div
                  aria-hidden="true"
                  class="fade tab-pane"
                  role="tabpanel"
                >
                  <div
                    class="row"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
